From f4e27fecc9592e66eda1e639b2c90efb2ae47fe1 Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Fri, 8 Nov 2019 17:48:40 -0500
Subject: [PATCH 30/60] kmssink: Ensure we have an allocator before importing

This fixes cases where the kms allocator API was called with a null
pointer.

(cherry picked from commit a83e0036ea66207e8ee264f944e1cfc5ebf1665e)
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 sys/kms/gstkmssink.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/sys/kms/gstkmssink.c b/sys/kms/gstkmssink.c
index 4095b35da..50c339a97 100644
--- a/sys/kms/gstkmssink.c
+++ b/sys/kms/gstkmssink.c
@@ -1398,6 +1398,8 @@ gst_kms_sink_import_dmabuf (GstKMSSink * self, GstBuffer * inbuf,
       return FALSE;
   }
 
+  ensure_kms_allocator (self);
+
   kmsmem = (GstKMSMemory *) gst_kms_allocator_get_cached (mems[0]);
   if (kmsmem) {
     GST_LOG_OBJECT (self, "found KMS mem %p in DMABuf mem %p with fb id = %d",
-- 
2.20.1

